### PR TITLE
RadioGroupButton: VR motion implementation

### PR DESCRIPTION
**Pull request by diyorbek**
### Summary

VR RadioGroupButton motion implementation WITHOUT other VR specific styles.

#### What changed?

**BEFORE**
ezgif-7-4b5c4062c9

**AFTER**
ezgif-1-a2e9bd38d7

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-8378)
- [Figma](https://www.figma.com/design/pMibC8csfCbNMU4jgLVUJh/Motion-Pilot-specs?node-id=2163-7303&t=rCSdnETB8qo1SoyQ-1)

### Checklist

- [ ] Added unit tests
- [ ] Added documentation + accessibility tests
- [x] Verified accessibility: keyboard & screen reader interaction
- [x] Checked dark mode, responsiveness, and right-to-left support
- [x] Checked stakeholder feedback (e.g. Gestalt designers, relevant feature teams)

### CODE REVIEW

File: packages/gestalt/src/ButtonToggle.tsx [hunk 1]
```diff
@@ -186,7 +186,7 @@ const ButtonToggleWithForwardRef = forwardRef<HTMLButtonElement, Props>(function
   const isDarkModeRed = isDarkMode && color === 'red';
   const { isFocusVisible } = useFocusVisible();
 
-  const buttonToggleAnimation = useButtonToggleAnimation();
+  const buttonToggleAnimation = useTapScaleAnimation();
```
Comment (AlbertCarreras) [inline packages/gestalt/src/ButtonToggle.tsx:189]: is this expected?

Comment (AlbertCarreras) [reply packages/gestalt/src/ButtonToggle.tsx:189]: just renaming i guess

File: packages/gestalt/src/RadioGroupButton.tsx [hunk 2]
```diff
@@ -130,27 +140,78 @@ const RadioGroupButtonWithForwardRef = forwardRef<HTMLInputElement, Props>(funct
...
+
   const styleSize = size === 'sm' ? controlStyles.sizeSm : controlStyles.sizeMd;
 
   const bgStyle = disabled && !checked ? styles.BgDisabled : styles.BgEnabled;
 
   const { isFocusVisible } = useFocusVisible();
 
   const { parentName } = useRadioGroupContext();
+
+  const tapScaleAnimation = useTapScaleAnimation();
```
Comment (AlbertCarreras) [inline packages/gestalt/src/RadioGroupButton.tsx:153]: nit  

 const { handleMouseDown, handleMouseUp } = useTapScaleAnimation();

File: packages/gestalt/src/RadioGroupButton.tsx [hunk 3]
```diff
@@ -130,27 +140,78 @@ const RadioGroupButtonWithForwardRef = forwardRef<HTMLInputElement, Props>(funct
...
+  const radioButtonStyles = isInVRExperiment
+    ? classnames(styleSize, layoutStyles.relative, borderStyles.circle, tapScaleAnimation.classes, {
+        [focusStyles.accessibilityOutlineFocus]: focused && isFocusVisible,
+      })
+    : classnames(bgStyle, borderColor, borderWidth, styleSize, styles.RadioButton, {
+        [focusStyles.accessibilityOutlineFocus]: focused && isFocusVisible,
+      });
+
+  const checkedStyles = classnames(
+    styles.VRRadioButton,
```
Comment (AlbertCarreras) [inline packages/gestalt/src/RadioGroupButton.tsx:170]: can we dry this so we know which props are reused in unchecked and checked and which ones are for one or the other.

   p.e. all this are shared:
styles.VRRadioButton,
borderColor,
 bgStyle,
    styleSize,

**Comment by AlbertCarreras** [comment]
i assume we don't need to listen to useReducedMotion here... can you confirm that?

**Review by AlbertCarreras — APPROVED** [review]
Few nits, but overall LGTM

File: packages/gestalt/src/ButtonToggle.tsx (hunk 1 above)
Comment (diyorbek) [reply packages/gestalt/src/ButtonToggle.tsx:189]: yes, i renamed it cuz now they is shared

File: packages/gestalt/src/RadioGroupButton.tsx (hunk 2 above)
Comment (diyorbek) [reply packages/gestalt/src/RadioGroupButton.tsx:153]: I didn't destructure the object to keeps its context "visible" while reading the code. We may have other even handler functions in the future. But this one is just for the animation. That's why i think its better to keep it this way.

File: packages/gestalt/src/RadioButton.css [hunk 4]
```diff
@@ -57,3 +57,35 @@
 .InputEnabled {
   composes: pointer from "./Cursor.css";
 }
+
```
Comment (rlingineni) [inline packages/gestalt/src/RadioButton.css:60]: Maybe add a comment here that these are VR related items

**Review by rlingineni — APPROVED** [review]

**Pull request merged**